module/reports/helper.js

```diff
@@ -0,0 +1,443 @@
+
+                let query = {
+                    "entityInformation._id": mongoose.Types.ObjectId(entityId),
+                }
+
+                let searchQuery = [];
+                if (search !== "") {
+                    searchQuery = [{ "programInformation.name": new RegExp(search, 'i') }];
+                }
+
+                const projectDocuments = await UserProjectsHelper.projects(
+                    query,
+                    pageSize,
+                    pageNo,
+                    searchQuery,
+                    ["programInformation", "entityInformation", "userId"]
+                );
+
+                if (projectDocuments.result && projectDocuments.result.count == 0) {
+                    return resolve({
+                        message: CONSTANTS.apiResponses.PROGRAMS_NOT_FOUND,
+                        result: []
+                    })
+                }
+
+                let programs = [];
+
+                let projectDetails = projectDocuments.result.data;
+                for (let index = 0; index < projectDetails.length; index++) {
+                    programs.push({
+                        name: projectDetails[index].programInformation.name,
+                        _id: projectDetails[index].programInformation._id
+                    });
+                }
+
+                return resolve({
+                    message: CONSTANTS.apiResponses.PROGRAMS_FOUND,
+                    result: {
+                        data: programs,
+                        count: projectDocuments.result.count
+                    }
+                });
+
+            } catch (error) {
+                return reject(error);
+            }
+        })
+    }
+
+
+
+    /**
+      * Get report types
+      * @method
+      * @name getTypes 
+      * @returns {Object} - returns report types
+     */
+
+    static getTypes() {
+        return new Promise(async (resolve, reject) => {
+            try {
+
+                let reportTypes = [
+                    {
+                        label: "Weekly",
+                        value: 0
+                    },
+                    {
+                        label: "Monthly",
+                        value: 1
+                    }, {
+                        label: "Quarterly",
+                        value: 3
+                    }]
+                resolve({ result: reportTypes, message: CONSTANTS.apiResponses.REPORT_TYPES_FOUND });
+
+            } catch (error) {
+                return reject(error);
+            }
+        })
+    }
+
+    /**
+    * Detail view Report.
+    * @method
+    * @name detailView 
+    * @param {String} entityId - mapped entity id.
+    * @param {String} userId - Logged in user id.
+    * @param {String} reportType - report type monthly or quterly.
+    * @param {String} programId - program id
+    * @returns {Object} - response consist of chart report data
+   */
+
+    static detailView(entityId="",userId,reportType,programId) {
```

Comment:
@rakeshSgr - Please use proper spaces and handle the scenario of not all different arguments are not passed.

Reply:
resolved